substrafl/experiment.py

```diff
@@ -309,6 +310,8 @@ def execute_experiment(
         clean_models=clean_models,
     )
 
+    dependencies.compute_in_cache_directory()
```

Comment:
The main reason is that the only way I thought about making this possible was to create a directory where to store all the computation. But we need this directory to be persistent enough to exists when registering the functions, and delete it when done. That is why I made that choice that creating this folder must be explicit so that the user think to delete it if he uses this function.

100% agree that it's a weird way to deal with this... Completely open if we have an other option !

Reply:
Correct me if I'm missing something, but we need to move the files for two reasons: 
1. because we need the selective copy mechanism before building the wheels
2. because `pip-tools` has this weird behaviour that is fine if you use it as a CLI, but cumbersome to integrate in a library. 

I think we can find some workaround the second point (like a tmp dir and storing the content of the requirements.txt file afterward), but point 1 is trickier.
But then, point 1 is only useful for creating the function archive and during the dependency compilation, we only need the content of the setup.py or the pyproject.toml anyway, so maybe there is a smart solution to find there 🤔

Reply:
(In all cases, if we end up using user-managed case, it's important to wrap it up in a `try: ... finally` clause I think)

Reply:
Yes but we need the files to persist while registering the functions :/ so we can't really wrap the tmp dir in that kind of clause :/

Reply:
Summary of call: we will use the `__del__` method provided by Python, called at the object deletion.